packages/rocketchat-livechat/server/lib/Livechat.js

```diff
@@ -363,13 +365,17 @@ RocketChat.Livechat = {
 		} else if (RocketChat.settings.get('Livechat_Routing_Method') !== 'Guest_Pool') {
 			agent = RocketChat.Livechat.getNextAgent(transferData.departmentId);
 		} else {
+			RocketChat.models.Rooms.changeDepartmentIdByRoomId(room._id, transferData.departmentId);
```

Comment:
If the `departmentId` changes, the new department needs to be updated in any of the routing methods, not just in the `Least_Amount` routing.
So, the right place to call this function is not here.